src/app/Modules/auth/register-page/register-page.component.ts

```diff
@@ -14,28 +15,46 @@ export class RegisterPageComponent implements OnInit {
   hide2:boolean = true;
   constructor(private fb: FormBuilder,
     private router: Router,
-    private aRouter: ActivatedRoute) { 
+    private userService: AuthService) { 
       this.userForm = this.fb.group({
         name: ['', Validators.required],
         lastname: ['', Validators.required],
-        username: ['', Validators.required],
+        username: ['', /* Validators.required */],
         password: ['', [
           Validators.required,
           Validators.minLength(6),
           Validators.maxLength(22)
         ]],
-        confirmpassword: ['', [
-          Validators.required, 
-          Validators.requiredTrue
+        confirmpassword: ['',[
+          Validators.required,
         ]],
         role: ['', Validators.required],
-      })
+      }, {validator: ConfirmedValidator('password', 'confirmpassword')})
```

Comment:
When the user starts typing the password, the field turns red immediately to indicate there's an error even though the user hasn't finished writing the password. This can be fixed by simply adding:

```
{
  validator: ConfirmedValidator('password', 'confirmpassword'), 
  updateOn:  'blur'
 }
```
`updateOn: "blur"` will make it so that the field is only updated once the user clicks on something else, and so the validation will be executed only when that occurs.